improver/nowcasting/forecasting.py

```diff
@@ -222,6 +226,90 @@ def point_in_bounds(x, y, nx, ny):
 
         return adv_field
 
+    @staticmethod
+    def _update_time(input_time, advected_cube, timestep):
+        """Increment validity time on the advected cube
+
+        Args:
+            input_time (iris.coords.Coord):
+                Time coordinate from source cube
+            advected_cube (iris.cube.Cube):
+                Cube containing advected data (modified in place)
+            timestep (datetime.timedelta)
+                Time difference between the advected output and the source
+        """
+        original_datetime = next(input_time.cells())[0]
+        new_datetime = original_datetime + timestep
+        new_time = (input_time.units).date2num(new_datetime)
+        advected_cube.coord("time").points = new_time
+        advected_cube.coord("time").convert_units(TIME_REFERENCE_UNIT)
+        advected_cube.coord("time").points = (
+            np.around(advected_cube.coord("time").points).astype(
+                TIME_REFERENCE_DTYPE))
+
+    @staticmethod
+    def _add_forecast_reference_time(input_time, advected_cube):
+        """Add or replace a forecast reference time on the advected cube"""
+        try:
+            advected_cube.remove_coord("forecast_reference_time")
+        except CoordinateNotFoundError:
+            pass
+
+        frt_coord = input_time.copy()
+        frt_coord.rename("forecast_reference_time")
+        frt_coord.convert_units(TIME_REFERENCE_UNIT)
+        frt_coord.points = np.around(frt_coord.points).astype(
+            TIME_REFERENCE_DTYPE)
+        advected_cube.add_aux_coord(frt_coord)
+
+    @staticmethod
+    def _add_forecast_period(advected_cube, timestep):
+        """Add or replace a forecast period on the advected cube"""
+        try:
+            advected_cube.remove_coord("forecast_period")
+        except CoordinateNotFoundError:
+            pass
+
+        forecast_period_seconds = np.int32(timestep.total_seconds())
+        forecast_period_coord = AuxCoord(forecast_period_seconds,
+                                         standard_name="forecast_period",
+                                         units="seconds")
+        advected_cube.add_aux_coord(forecast_period_coord)
+
+    def _create_output_cube(self, cube, advected_data, timestep):
```

Comment:
I don't think any of the unit tests actually check the attributes on the output cubes. I would prefer to have unit tests for all the new functions you've factored out, but if we aren't going to do unit tests for all the private functions I think it would be useful to at least have unit tests to cover these attribute changes. What do you think?

Reply:
There are loads of tests in `test_AdvectField.py::Test_process` that cover attribute values.  Were you expecting to find them somewhere different?

Reply:
Sorry, I think I was looking in the wrong place. I think you're right that the essential stuff is covered by the process unit tests for AdvectField.